src/components/common/CustomCheckbox.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="custom-checkbox__wrapper">
+    <input
+      class="custom-checkbox__input"
+      type="checkbox"
+      value="agreed"
+      :id="name"
+      :name="name"
+      v-model="value"
```

Comment:
You should only bind the value here instead and do (as is stated in the docs):

```suggestion
      :checked="modelValue"
```
You already have the `@change` handler so that's all that is truly needed now 😄 

Also, now we do not need the extra `value` ref 😉